test/lib/jdk/test/lib/hprof/parser/Reader.java

```diff
@@ -94,19 +93,39 @@ public static Snapshot readFile(String heapFile, boolean callStack,
                     = new HprofReader(heapFile, in, dumpNumber,
                                       callStack, debugLevel);
                 return r.read();
-            } else if ((access = GzipRandomAccess.getAccess(heapFile, 16)) != null) {
+            } else if ((i >>> 8) == GZIP_HEADER_MAGIC) {
+                // Possible gziped file, try decompress it and get the stack trace.
                 in.close();
-                try (BufferedInputStream gzBis = new BufferedInputStream(access.asStream(0));
-                     PositionDataInputStream pdin = new PositionDataInputStream(gzBis)) {
+                String deCompressedFile = "heapdump" + System.currentTimeMillis() + ".hprof";
```

Comment:
Is it necessary to create a file with the decompressed output rather than just stream the decompressed output to a FileInputStream?

Reply:
Correct! seems FileInputStream is enough, I will make the change. Thanks!

Reply:
Hi @plummercj, 
Sorry for late response, I tried to avoid using the decompressed file, but it seems not simple as I expected. 
The reason is that at line 121, the HprofReader require the filename and it use this file to create a ReadBuffer internally, and a RandomAccessFile will be created using the filename. 
So I think may be it is not easy to make the change here, and maybe we could track it using a new bug if necessary, which may change the HprofReader implementation.
What do you think?

Reply:
Yes, it's fine then the way it currently is implemented. Thanks.